skimage/feature/texture.py

```diff
@@ -257,15 +261,29 @@ def graycoprops(P, prop='contrast'):
         results = np.sqrt(asm)
     elif prop == 'ASM':
         results = np.sum(P ** 2, axis=(0, 1))
-    elif prop == 'correlation':
+    elif prop in ['correlation', 'mean', 'variance']:
         results = np.zeros((num_dist, num_angle), dtype=np.float64)
         I = np.array(range(num_level)).reshape((num_level, 1, 1, 1))
         J = np.array(range(num_level)).reshape((1, num_level, 1, 1))
-        diff_i = I - np.sum(I * P, axis=(0, 1))
-        diff_j = J - np.sum(J * P, axis=(0, 1))
 
-        std_i = np.sqrt(np.sum(P * (diff_i) ** 2, axis=(0, 1)))
-        std_j = np.sqrt(np.sum(P * (diff_j) ** 2, axis=(0, 1)))
+        # it does not matter if we use i or j
+        mean_i = np.sum(I * P, axis=(0, 1))
+        if prop == 'mean':
+            return mean_i
```

Comment:
I really don't like that new return statements are introduced into this already large function. I don't see an immediate way to simplify this without unnecessarily computing the correlation in some cases.

Reply:
I agree with you. Honestly, the reason why I put this return there is because I would like to avoid useless computation. Waiting for feedback from others.

Reply:
Note that mean_i is computed unnecessarily also.

The interface for this function is not great. It would be nice to be able to compute multiple props, also as to re-use intermediate results along the way.

We could, without API breakage, introduce something like this:

```python
res = graycoprops(..., ['variance', 'mean'])
res['mean'] -> 2D array
res['variance'] -> 2D array
```

```python
res = graycoprops(..., 'mean')
res -> 2D array
```

Reply:
Honestly, I like your suggestion because I see other libraries doing the same. Do you want me to propose something like that?

Reply:
If we want to make the API consistent, we can also error out if the properties are not given as a list.

The dataclass pattern looks fine, as long as a person doesn't have to carry along a lot of data to compute the members (which, unfortunately, in the case of GLCM may be the case).